lib/active_record/connection_adapters/sqlserver_adapter.rb

```diff
@@ -194,7 +196,7 @@ def reset!
 
       def tables_with_referential_integrity
         schemas_and_tables = select_rows <<-SQL.strip_heredoc
-          SELECT s.name, o.name
+          SELECT s.name sc, o.name tab
```

Comment:
The method handle.all which is used to retrieve the result within DatabaseStatements#handle_to_names_and_values_jdbc returns a hash with {column: value}.
Without the aliases there would only be one column.